contracts/templates/ERC20Template3.sol

```diff
@@ -0,0 +1,1184 @@
+pragma solidity 0.8.12;
+// Copyright BigchainDB GmbH and Ocean Protocol contributors
+// SPDX-License-Identifier: (Apache-2.0 AND CC-BY-4.0)
+// Code is Apache-2.0 and docs are CC-BY-4.0
+
+import "../interfaces/IERC721Template.sol";
+import "../interfaces/IERC20Template.sol";
+import "../interfaces/IFactoryRouter.sol";
+import "../interfaces/IFixedRateExchange.sol";
+import "../interfaces/IDispenser.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "../utils/ERC20Roles.sol";
+
+/**
+ * @title DatatokenTemplate
+ *
+ * @dev ERC20Template3 is an ERC20 compliant token template
+ *      Used by the factory contract as a bytecode reference to
+ *      deploy new Datatokens.
+ * IMPORTANT CHANGES:
+ *  - creation of pools/dispensers is not allowed
+ *  - creation of additional fixed rates is not allowed (only one can be created)
+ */
+contract ERC20Template3 is
+    ERC20("test", "testSymbol"),
+    ERC20Roles,
+    ERC20Burnable,
+    ReentrancyGuard
+{
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    string private _name;
+    string private _symbol;
+    uint256 private _cap;
+    uint8 private constant _decimals = 18;
+    bool private initialized = false;
+    address private _erc721Address;
+    address private paymentCollector;
+    address private publishMarketFeeAddress;
+    address private publishMarketFeeToken;
+    uint256 private publishMarketFeeAmount;
+    uint256 private rate = 0;
+    uint256 public constant BASE = 1e18;
+
+    // -------------------------- PREDICTOOR --------------------------
+    enum Status {
+        Pending,
+        Paying,
+        Canceled
+    }
+    event PredictionSubmitted(
+        address indexed predictoor,
+        uint256 indexed slot,
+        uint256 stake
+    );
+    event PredictionPayout(
+        address indexed predictoor,
+        uint256 indexed slot,
+        uint256 stake,
+        uint256 payout,
+        bool prediction,
```

Comment:
Rename to `predval`

Why: disambiguate from each predictoor's `predval`, versus the top-level aggregated predval `agg_predval`.